pages/app-layout/utils/iframe-wrapper.tsx

```diff
@@ -0,0 +1,41 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useContext, useEffect, useRef } from 'react';
+import ReactDOM from 'react-dom';
+import AppContext from '../../app/app-context';
+import styles from './iframe-wrapper.scss';
+
+function copyStyles(srcDoc: Document, targetDoc: Document) {
+  for (const stylesheet of Array.from(srcDoc.querySelectorAll('link[rel=stylesheet]'))) {
+    const newStylesheet = targetDoc.createElement('link');
+    for (const attr of stylesheet.getAttributeNames()) {
+      newStylesheet.setAttribute(attr, stylesheet.getAttribute(attr)!);
+    }
+    targetDoc.head.appendChild(newStylesheet);
+  }
+}
+
+export function IframeWrapper({ id, AppComponent }: { id: string; AppComponent: React.ComponentType }) {
+  const { urlParams } = useContext(AppContext);
+  const iframeRef = useRef<HTMLIFrameElement>(null);
+
+  useEffect(() => {
+    const iframeEl = iframeRef.current;
+    if (!iframeEl) {
+      return;
+    }
+    console.log('Mounting iframe');
+    const innerAppRoot = iframeEl.contentDocument!.createElement('div');
+    iframeEl.contentDocument!.body.appendChild(innerAppRoot);
+    copyStyles(document, iframeEl.contentDocument!);
+    if (urlParams.visualRefresh) {
+      iframeEl.contentDocument!.body.className = 'awsui-visual-refresh';
```

Comment:
also dark mode class? currently it doesn't work inside iframe https://d21d5uik3ws71m.cloudfront.net/components/a560afc1ed48ad70d78a9b2b8abe3051b575ee2e/dev-pages/index.html#/dark/app-layout/multi-layout-iframe/

Reply:
Good find, fixed this

Reply:
what about direction attribute?
it also looks wrong in iframes https://d21d5uik3ws71m.cloudfront.net/components/93c50d59ab8dd23fa3c3a75bc24c4d75ea8fa92b/dev-pages/index.html#/dark/app-layout/multi-layout-iframe/?density=compact&motionDisabled=true&direction=rtl

Reply:
Fixed this one as well. Did not use a mutation observer there, because we do not support changing it dynamically anyway.

Also looked at other global flags, I think now I propagated everything we set in this file: https://github.com/cloudscape-design/components/blob/main/pages/app/index.tsx